contents/docs/experiments/adding-experiment-code.mdx

```diff
@@ -0,0 +1,218 @@
+---
+title: Adding experiment code
+sidebar: Docs
+showTitle: true
+availability:
+    free: none
+    selfServe: full
+    enterprise: full
+---
+
+Once you've created your experiment in the PostHog dashboard, the next step is to add your code:
+
+## Step 1: Fetch the feature flag
+
+In your experiment, each user is randomly assigned to a variant (usually either 'control' or 'test'). To check which variant a user has been assigned to, you'll fetch the experiment feature flag. You can then customize their experience based on the value in the feature flag:
+
+<MultiLanguage>
+
+```js-web
+// Ensure flags are loaded before usage.
+// You'll only need to call this on the code the first time a user visits.
+// See this doc for more details: https://posthog.com/docs/feature-flags/manual#ensuring-flags-are-loaded-before-usage
+posthog.onFeatureFlags(function() {
+    // feature flags should be available at this point
+    if (posthog.getFeatureFlag('experiment-feature-flag-key')  == 'variant-name') {
+        // do something
+    }
+})
+
+// Otherwise, you can just do:
+if (posthog.getFeatureFlag('experiment-feature-flag-key')  == 'variant-name') {
+    // do something
+}
+
+// You can also test your code by overriding the feature flag:
+// e.g., posthog.feature_flags.override({'experiment-feature-flag-key': 'test'})
+```
+
+```react
+// You can either use the `useFeatureFlagVariantKey` hook,
+// or you can use the feature flags component - https://posthog.com/docs/libraries/react#feature-flags-react-component
+
+// Method one: using the useFeatureFlagVariantKey hook
+import { useFeatureFlagVariantKey } from 'posthog-js/react'
+
+function App() {
+    const variant = useFeatureFlagVariantKey('experiment-feature-flag-key')
+    if (variant == 'variant-name') {
+        // do something
+    }
+}
+
+// Method two: using the feature flags component
+import { PostHogFeature } from 'posthog-js/react'
+
+function App() {
+
+    return (
+        <PostHogFeature flag='experiment-feature-flag-key' match={'variant-name'}>
+            <div>
+                <!-- the component to show -->
+            </div>
+        </PostHogFeature>
+    )
+}
+
+
+// You can also test your code by overriding the feature flag:
+// e.g., posthog.feature_flags.override({'experiment-feature-flag-key': 'test'})
```

Comment:
This doesn't work in React.

Reply:
Are you sure? Just tested it in my nextjs project and it works